### ISSUE TITLE
StringShouldBeRawString false positive with regex

### ISSUE THREAD

**Issue by WonderCsabo** [detekt/detekt#6221]
## Expected Behavior
The `StringShouldBeRawString` rule should not report code with regex strings.

## Observed Behavior
The `StringShouldBeRawString` rule does report code with regex strings.

## Steps to Reproduce
The following code snippet is flagged by the rule:

```kotlin
val AlphaNumericWithSpecialCharacters =
    "[a-zA-Z0-9 áéíóöőúüűÁÉÍÓÖŐÚÜŰ&„’\"”'._+–()/:;,@\\[\\]~{}!%=˝×€$*?-]*".toRegex()
```

## Your Environment
* Version of detekt used: 1.23.0

**Comment by WonderCsabo** [comment]
Ah i see, thanks!
